addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRuleUnitTest.java

```diff
@@ -82,4 +98,107 @@ void shouldReturnExpectedExampleAlert() {
         assertThat(alert.getRisk(), is(equalTo(Alert.RISK_HIGH)));
         assertThat(alert.getConfidence(), is(equalTo(Alert.CONFIDENCE_MEDIUM)));
     }
+
+    @Test
+    void shouldFindGitRepoExposed() throws Exception {
+        // Given
+        this.nano.addHandler(new GitServerHandler());
+
+        HttpSender realSender = new HttpSender(HttpSender.ACTIVE_SCANNER_INITIATOR);
+        when(parent.getHttpSender()).thenReturn(realSender);
+
+        String url = "http://127.0.0.1:" + nano.getListeningPort() + "/custom/Target.java";
+        HttpMessage msg = new HttpMessage(new URI(url, true));
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\nContent-Type: text/html\r\n\r\n");
+
+        rule.init(msg, parent);
+        rule.setAttackStrength(AttackStrength.HIGH);
```

Comment:
Should be removed, it's irrelevant for the test case (the base message is a non-404).